vsock-bridge/src/socket.cpp

```diff
@@ -188,18 +194,14 @@ namespace vsockio
 	void Socket::closeInput()
 	{
 		_inputClosed = true;
-	}
 
-	void Socket::closeOutput()
-	{
-		if (_inputClosed)
-		{
-			Logger::instance->Log(Logger::DEBUG, "shutdown 6");
-			shutdown();
-		}
+		// There may be a termination buffer queued by the peer. Poller may not be able to detect that and mark
+		// the socket as ready for write in a timely fashion. Force process the queue now.
+		_outputReady = true;
+		writeToOutput();
```

Comment:
I don't get why there is concern if a termination buffer is queued by peer here.

Reply:
You are right. Force processing the queue from `onPeerClosed()` is sufficient.